### DISCUSSION TITLE
How to get Infeasible Solution

### DISCUSSION THREAD

**Discussion by aihiangng** [google/or-tools#3250] [CP-SAT questions]
2 questions

- Is it possible to get the constraints that led to infeasible solution in Python?
- Is it possible to get the infeasible solution for CP-SAT/MIP ? With this infeasible solution I can write a solution checker to identify constraints not met. 

Thanks!

**Comment by lperron** [comment]
1) yes. But it is  convoluted. See the assumptions_sat examples
2) I do not understand. a) CP-SAT has a solution checker built-in. b) it should never return an infeasible solution.

**Reply by aihiangng** [reply]
CPSat will return solver statusbas Infeasible. Is there anyway to extract the Infeasible solution?

**Reply by lperron** [reply]
CP-SAT never builds an infeasible solution internally/
